lib/AdminPages/Pages/Newsletters.php

```diff
@@ -102,6 +108,11 @@ public function render() {
     $data['roles'] = $wp_roles->get_names(); // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
     $data['roles']['mailpoet_all'] = $this->wp->__('In any WordPress role', 'mailpoet');
 
+    $mssActive = Bridge::isMPSendingServiceEnabled();
+    $mssKeyValid = $this->servicesChecker->isMailPoetAPIKeyValid();
+    $mssKeyPendingApproval = $this->settings->get('mta.mailpoet_api_key_state.data.is_approved') === false;
+    $data['mss_key_pending_approval'] = $mssActive && $mssKeyValid && $mssKeyPendingApproval;
```

Comment:
I can see this exact same logic in two places. What about putting it into service checker service method?

Reply:
Good idea, done.